tests/aws/services/s3/test_s3.py

```diff
@@ -964,31 +978,132 @@ def test_create_bucket_via_host_name(self, s3_vhost_client, aws_client, region_n
             s3_vhost_client.delete_bucket(Bucket=bucket_name)
 
     @markers.aws.validated
-    def test_put_and_get_bucket_policy(self, s3_bucket, snapshot, aws_client, allow_bucket_acl):
+    def test_get_bucket_policy(self, s3_bucket, snapshot, aws_client, allow_bucket_acl, account_id):
+        snapshot.add_transformer(snapshot.transform.key_value("Resource"))
+        snapshot.add_transformer(snapshot.transform.key_value("BucketName"))
+
+        with pytest.raises(ClientError) as e:
+            aws_client.s3.get_bucket_policy(Bucket=s3_bucket)
+        snapshot.match("get-bucket-policy-no-such-bucket-policy", e.value.response)
+
+        policy = _simple_bucket_policy(s3_bucket)
+        aws_client.s3.put_bucket_policy(Bucket=s3_bucket, Policy=json.dumps(policy))
+
+        # retrieve and check policy config
+        response = aws_client.s3.get_bucket_policy(Bucket=s3_bucket)
+        snapshot.match("get-bucket-policy", response)
+        assert policy == json.loads(response["Policy"])
+
+        response = aws_client.s3.get_bucket_policy(Bucket=s3_bucket, ExpectedBucketOwner=account_id)
+        snapshot.match("get-bucket-policy-with-expected-bucket-owner", response)
+        assert policy == json.loads(response["Policy"])
+
+        with pytest.raises(ClientError) as e:
+            aws_client.s3.get_bucket_policy(Bucket=s3_bucket, ExpectedBucketOwner="000000000002")
```

Comment:
this is something you could not really guess, but we run a pipeline with multi-account setup in mind, and it could be than this test is ran with the `000000000002` account id as the account id of the main "test" account, and this test then wouldn't fail. 

I think there's a fixture called `secondary_account_id` that will always be different than `account_id`, allowing you to properly validate it while making sure it never fails in our pipeline.